qir/qat/Apps/Qat/ProfileConfiguration.hpp

```diff
@@ -0,0 +1,292 @@
+    configuration_manager.updateParameter("disable-record-output-support", false);
+
+    configuration_manager.updateParameter("unroll-loops", true);
+    configuration_manager.updateParameter("allow-partial", true);
+    configuration_manager.updateParameter("allow-peeling", true);
+    configuration_manager.updateParameter("allow-runtime", true);
+    configuration_manager.updateParameter("allow-upper-bound", true);
+    configuration_manager.updateParameter("allow-profile-based-peeling", true);
+    configuration_manager.updateParameter<uint64_t>("full-unroll-count", 1024);
+    configuration_manager.updateParameter<int32_t>("unroll-opt-level", 3);
+    configuration_manager.updateParameter("only-when-forced", false);
+    configuration_manager.updateParameter("forget-scev", false);
+    configuration_manager.updateParameter("always-inline", true);
+    configuration_manager.updateParameter<int32_t>("inlining-parameter", 2147483647);
+    configuration_manager.updateParameter("eliminate-constants", true);
+    configuration_manager.updateParameter("eliminate-dead-code", true);
+    configuration_manager.updateParameter("eliminate-memory", true);
+
+    configuration_manager.updateParameter("delete-dead-code", false);
+    configuration_manager.updateParameter("clone-functions", false);
+    configuration_manager.updateParameter("transform-execution-path-only", false);
+    configuration_manager.updateParameter<uint64_t>("max-recursion", 1);
+    configuration_manager.updateParameter("assume-no-except", false);
+    configuration_manager.updateParameter("reuse-qubits", true);
+    configuration_manager.updateParameter("reuse-results", false);
+
+    configuration_manager.updateParameter("lower-switch", true);
+    configuration_manager.updateParameter("should-eleminate-zext-i1", true);
+    configuration_manager.updateParameter("pull-records-back", false);
+
+    configuration_manager.updateParameter("annotate-qubit-use", true);
+    configuration_manager.updateParameter("annotate-result-use", true);
+    configuration_manager.updateParameter("annotate-max-qubit-index", true);
+    configuration_manager.updateParameter("annotate-max-result-index", true);
+    configuration_manager.updateParameter("reindex-qubits", true);
+    configuration_manager.updateParameter("replace-qubit-on-reset", false);
+    configuration_manager.updateParameter("inline-after-id-change", true);
+
+    configuration_manager.updateParameter("separate-circuits", false);
+
+    configuration_manager.updateParameter("allow-internal-calls", false);
+    configuration_manager.updateParameter("requires-qubits", true);
+    configuration_manager.updateParameter("requires-results", true);
+    configuration_manager.updateParameter("allow-poison", false);
+    configuration_manager.updateParameter("allow-undef", false);
+}
+
+inline void configureProviderb340Profile(ConfigurationManager& configuration_manager)
```

Comment:
It is not something that I typically see when such a large function 
* is defined in a header rather than a .cpp file; each .cpp file that includes this header may get a separate copy of this function body during compilation, and those copies may stay not merged into a single copy during linking;
* is recommended to be inlined (`inline`).

Reply:
Done.